src/wrappers/foresight/TestRunnerSupport.ts

```diff
@@ -0,0 +1,193 @@
+ */
+
+const sendTestRunStatus = async () => {
+    if (!testRunScope || !initialized) {
+
+        ThundraLogger.debug('<ThundraRunnerSupport> Test run not started. TestRunStatus will not send.');
+        return;
+    }
+
+    const config = ConfigProvider.thundraConfig;
+    const { apiKey } = config;
+
+    const statusTimestamp = new Date().getTime();
+    const testRunStatus = TestRunStatus
+        .builder()
+            .withId(testRunScope.id)
+            .withProjectId(projectId)
+            .withTaskId(testRunScope.taskId)
+            .withStartTimestamp(testRunScope.startTimestamp)
+            .withStatusTimestamp(statusTimestamp)
+            .withTotalCount(testRunScope.context.totalCount)
+            .withSuccessfulCount(testRunScope.context.successfulCount)
+            .withFailedCount(testRunScope.context.failedCount)
+            .withIgnoredCount(testRunScope.context.ignoredCount)
+            .withAbortedCount(testRunScope.context.abortedCount)
+            .withHostName(hostName)
+            .withEnvironmentInfo(EnvironmentSupport.getEnvironmentInfo())
+        .build();
+
+    try {
+        await wrapperContext.reporter.sendReports([Utils.generateReport(testRunStatus, apiKey)]);
+    } catch (error) {
+        ThundraLogger.error('<ThundraRunnerSupport> Test run status event did not sent.', error);
+    } finally {
+        startTestRunStatusEvent();
+    }
+};
+
+export const startTestRunStatusEvent = () => {
+
+    if (testStatusReportFreq) {
+        finishTestRunStatusEvent();
+        testRunStatusWork = setTimeout(sendTestRunStatus, testStatusReportFreq);
+    }
+};
+
+export const finishTestRunStatusEvent = () => {
+
+    if (testRunStatusWork) {
+        clearInterval(testRunStatusWork);
+        testRunStatusWork = null;
+    }
+};
+
+export const startTestRun = (): TestRunStart => {
+
+    const testRunId = getTestRunId();
+    const taskId = Utils.generateId();
+    const startTimestamp = new Date().getTime();
+
+    testRunScope = new TestRunScope(
+        testRunId,
+        taskId,
+        startTimestamp,
+        new TestRunContext(),
+    );
+
+    initialized = true;
+
+    return TestRunStart
+        .builder()
+            .withId(testRunId)
+            .withProjectId(projectId)
+            .withTaskId(taskId)
+            .withStartTimestamp(startTimestamp)
+            .withHostName(hostName)
+            .withEnvironmentInfo(EnvironmentSupport.getEnvironmentInfo())
+        .build();
+};
+
+export const finishTestRun = (): TestRunFinish => {
+    if (!initialized || !testRunScope) {
+
+        ThundraLogger.debug('<ThundraRunnerSupport> Test run not started. TestRunFinish will not send.');
+        return;
+    }
+
+    const finishTimestamp = new Date().getTime();
+
+    const {
+        totalCount,
+        successfulCount,
+        failedCount,
+        ignoredCount,
+        abortedCount,
+    } = testRunScope.context;
+
+    return TestRunFinish.
```

Comment:
clear (set `null`) `testRunScope`?

Reply:
clearTestRun method created and called after global teardown process